### ISSUE TITLE
tags

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#318]
Create tags that can be used to limit entrants to tournaments.

This will be used for BARGE. I mostly know what I want to do here, but haven't written anything down 'cause Silverton.

**Comment by ctm** [comment]
Heh. I wound up doing #339 for BARGE, in part because although this is easy, #339 was easier.

Now I'd like to at least explore this. I think it's basically:

- [ ] add tags table to database
- [ ] add players_tags join table to database
- [ ] add events_tags join table to database
- [ ] pre-populate the above tables with a barge_2020 tag and appropriate joins
- [ ] wherever mb2 passes down the barge column, pass a Vec of tags
- [ ] wherever mb2 checks against the barge column, check against the Vec of tags
- [ ] when all the above is working, add a migration that gets rid of the barge columns

After al of that is done I can then do something similar with the demo columns, although I'll need a separate demo tag for each concurrent demo, because we don't want two demos interfering with each other.

**Comment by ctm** [comment]
Time to read about [diesel associations](https://docs.diesel.rs/1.4.x/diesel/associations/index.html).

**Comment by ctm** [comment]
I spent a long time fighting with diesel, but couldn't figure out how to use a `GROUP BY` clause and get back a rows that contained `Vec`s. For the data we have, the difference is really meaningless, but I was learning other things as I went, so I kept pushing.

Anyway, the code is now written. I will deploy it either between this evening's game an tonight's game or after tonight's game (although that could be when I get up in the morning).

I spent more time than I would have liked on this, but it's nice to get it done.
